src/ChildProcess.php

```diff
@@ -50,25 +50,39 @@ public function all(): array
 
         return $hydrated;
     }
-
+        
     public function start(
+        string|array $cmd
         string $alias,
-        array $cmd,
         ?string $cwd = null,
         ?array $env = null,
         bool $persistent = false
     ): static {
+        $cwd = $cwd ?? base_path();
+
+        if (is_string($cmd)) {
+            // When a string is passed, explode it on the space
+            $cmd = array_values(array_filter(explode(' ', $cmd)));
```

Comment:
I'm not sure this will work well enough... How does Symfony Process/Laravel's Process facade handle this? Perhaps we can borrow their logic?

Reply:
Agreed. That might also provide a solution to the previously mentioned issue when passing strings instead of an array.

 I'll pick this up here tomorrow.

Reply:
Note on escaping paths containing spaces:

It seems Laravel's Process facade doesn't account for this either. When passing a path to a executable with a space as a string I get the following error.

``` php 
Process::run(PHP_BINARY . ' stdout.php')
    ->throw()
    ->output();
```

``` 
The command "/Users/gwleuverink/Library/Application Support/Herd/bin/php83 stdout.php" failed.

Exit Code: 127

Error Output:
================
sh: /Users/gwleuverink/Library/Application: No such file or directory
```

But when I pass it an array it knows to escape them individually:

``` php
Process::run([PHP_BINARY, 'stdout.php'])
    ->throw()
    ->output();
```

```
OK!0
```

Note this is expected output from the `stdout.php` script

------------

Next, I sourcedove Symfony process. When I dump the the `$commandline` variable right before being passed to `proc_open` here:

https://github.com/symfony/symfony/blob/6161368d792b72b313c25e4706ac1b09107f2193/src/Symfony/Component/Process/Process.php#L360

``` php 
Process::run([PHP_BINARY, 'stdout.php']);
// passed to proc_open: ["/Users/gwleuverink/Library/Application Support/Herd/bin/php83", "stdout.php"]

Process::run(PHP_BINARY . ' stdout.php');
// passed to proc_open: ["/Users/gwleuverink/Library/Application Support/Herd/bin/php83 stdout.php"]
```
I see the expected behaviour when passing it as a array. But when passing a string it doesn't explode on a space, but simply wraps it in an array.

## Conclusion

So in the end it looks like the behaviour escaping spaces already works the same as Laravel's process facade, handled on electron's side. Do you think we should add this caveat to the docs? @simonhamp 

How a string argument is passed however is a bit different. It is not exploded at al rather passed as a array with a single item. I hope electron handles this in a similar fashion under the hood. Shall I change this so it's passed the same way?

Reply:
Let's see what happens if we just wrap it in an array - this may mean we don't need to do any manipulation right, and could just leave it to Electron?

Reply:
Hope so. Will test tonight 👌🏻